src/client/providers/setInterpreterProvider.ts

```diff
@@ -75,21 +92,36 @@ function suggestPythonPaths(): Promise<vscode.QuickPickItem[]> {
 }
 
 function setPythonPath(pythonPath: string) {
-    // Waiting on https://github.com/Microsoft/vscode/issues/1396
-    // For now, just let the user copy this to clipboard
-    const copy_msg =  "Copy to Clipboard"
-
-    // If the user already has .vscode/settings.json in the workspace
-    // open it for them
     vscode.workspace.openTextDocument(workspaceSettingsPath())
-        .then(doc => vscode.window.showTextDocument(doc));
-    
-    vscode.window.showInformationMessage(pythonPath, copy_msg)
-        .then(item => {
-            if (item === copy_msg) {
-                ncp.copy(pythonPath)
+        .then(doc => {
+            const settingsText = doc.getText();
+            if (settingsText.search(REPLACE_PYTHONPATH_REGEXP) === -1) {
+                // Can't find the setting to replace - will just have to offer a copy button and instruct them to edit themselves.
+                openWorkspaceSettings().then(() => {
+                    const copyMsg =  "Copy to Clipboard"
+                    const newEntry = `"python.pythonPath": "${pythonPath}"`;
+                    vscode.window.showInformationMessage(`Please add an entry: ${newEntry}`, copyMsg)
+                        .then(item => {
+                            if (item === copyMsg) {
+                                ncp.copy(newEntry)
+                            }
+                        })
+                })
+            } else {
+                // Great, the user already has a setting stated that we can relibly replace!
+                const newSettingsText = settingsText.replace(REPLACE_PYTHONPATH_REGEXP, `$1"${pythonPath}"`);
+                replaceContentsOfFile(doc, newSettingsText).then(
```

Comment:
@jabooth  I think a better solution would be:
- Identify the line numbers in the text document containing the python.pythonPat and the values
- If such a value exists and can be found, then we replace it using the standard vscode.workspace.applyEdit...
- However if not found, then we inform the user about this, perhaps such an entry doesn't exist (vanilla setting.json file) or they have a setting such as follows:
```json
"python": {
    "pythonPath":"xyz...abc",

}
```

This would be much better and more efficient than replacing the entire text document. I know i've had issues in replacing entire text documents and that can stuff up the editor (bookmarks, breakpoints, etc).